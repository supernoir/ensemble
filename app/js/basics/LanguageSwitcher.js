import React from 'react';

import { Dropdown, Flag } from 'semantic-ui-react';
import intl from 'react-intl-universal';
import LOCALES from '../constants/locales';

export default class LanguageSwitcher extends React.Component {

	handleLocaleChange = (evt, locale) => {
		evt.preventDefault();
		this.props.setCurrentLocale(locale);
	}

	render(){
		return (
			<Dropdown item icon="globe">
				<Dropdown.Menu>
					{
						LOCALES.map((entry, index) => {
							return <Dropdown.Item
								key={entry.locale + '-' + index}
								value={entry.locale}
								onClick={(evt)=>this.handleLocaleChange(evt, entry.locale)}
							>
								<Flag name={entry.code} />{intl.get(entry.name)}
							</Dropdown.Item>;
						})
					}
				</Dropdown.Menu>
			</Dropdown>
		);
	}
}