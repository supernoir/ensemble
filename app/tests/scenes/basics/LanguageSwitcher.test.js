import React from 'react';
import { shallow } from 'enzyme';

const testData = {
	setCurrentLocale: jest.fn()
};

// Import Components
import LanguageSwitcher from '../../../js/basics/LanguageSwitcher';

test('LanguageSwitcher -- Snapshot Test', () => {
	const wrapper = shallow(<LanguageSwitcher
		setCurrentLocale={() => testData.setCurrentLocale()}
	/>
	);
	expect(wrapper).toMatchSnapshot();
});
