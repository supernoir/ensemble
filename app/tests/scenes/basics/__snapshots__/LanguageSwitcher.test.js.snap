// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher -- Snapshot Test 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  closeOnBlur={true}
  closeOnEscape={true}
  deburr={false}
  icon="globe"
  item={true}
  minCharacters={1}
  noResultsMessage="No results found."
  openOnFocus={true}
  renderLabel={[Function]}
  searchInput="text"
  selectOnBlur={true}
  selectOnNavigation={true}
  wrapSelection={true}
>
  <DropdownMenu>
    <DropdownItem
      key="en-US-0"
      onClick={[Function]}
      value="en-US"
    >
      <Flag
        as="i"
        name="us"
      />
    </DropdownItem>
    <DropdownItem
      key="es-ES-1"
      onClick={[Function]}
      value="es-ES"
    >
      <Flag
        as="i"
        name="es"
      />
    </DropdownItem>
    <DropdownItem
      key="de-DE-2"
      onClick={[Function]}
      value="de-DE"
    >
      <Flag
        as="i"
        name="de"
      />
    </DropdownItem>
  </DropdownMenu>
</Dropdown>
`;
